proxy.go

```diff
@@ -0,0 +1,292 @@
+package tunnel
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"sync"
+
+	"github.com/koding/logging"
+	"github.com/koding/tunnel/proto"
+)
+
+// Proxy is responsible for forwarding remote connection to local server and writing the response back.
+type Proxy interface {
+	Proxy(remote net.Conn, msg *proto.ControlMessage)
+}
+
+var (
+	httpLog = logging.NewLogger("http")
+	tpcLog  = logging.NewLogger("tcp")
+)
+
+// ProxyHTTP is Proxy implementation focused on forwarding HTTP traffic.
+//
+// When tunnel server requests a connection it's proxied to 127.0.0.1:incomingPort
+// where incomingPort is control message LocalPort.
+// Usually this is tunnel server's public exposed Port.
+// This behaviour can be changed by setting LocalAddr or FetchLocalAddr.
+// FetchLocalAddr takes precedence over LocalAddr.
+//
+// When connection to local server cannot be established proxy responds with http error message.
+type ProxyHTTP struct {
```

Comment:
Please make it `HTTPProxy`, as `ReverseProxy`, `OtherProxy` etc.

`ProxyHTTP` was a name for a function.